experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/types.ts

```diff
@@ -42,58 +42,101 @@ export interface SharedTreeList<
 	 * @param value - The content to insert.
-	 * @throws Throws if any of the input indices are invalid.
 	 */
 	insertAtStart(value: Iterable<ProxyNodeUnion<TTypes>>): void;
 
 	/**
-	 * Inserts new item(s) at the end of the sequence.
+	 * Inserts new item(s) at the end of the list.
 	 * @param value - The content to insert.
-	 * @throws Throws if any of the input indices are invalid.
 	 */
 	insertAtEnd(value: Iterable<ProxyNodeUnion<TTypes>>): void;
 
 	/**
 	 * Removes the item at the specified location.
 	 * @param index - The index at which to remove the item.
-	 * @throws Throws if any of the input indices are invalid.
+	 * @throws Throws if `index` is not in the range [0, `list.length`).
 	 */
 	removeAt(index: number): void;
 
 	/**
 	 * Removes all items between the specified indices.
-	 * @param start - The starting index of the range to remove (inclusive). Defaults to the start of the sequence.
+	 * @param start - The starting index of the range to remove (inclusive). Defaults to the start of the list.
 	 * @param end - The ending index of the range to remove (exclusive).
-	 * @throws Throws if any of the input indices are invalid.
-	 * If `end` is not supplied or is greater than the length of the sequence, all items after `start` are deleted.
+	 * @throws Throws if `start` is not in the range [0, `list.length`).
+	 * @throws Throws if `end` is less than `start`.
+	 * If `end` is not supplied or is greater than the length of the list, all items after `start` are deleted.
 	 */
 	removeRange(start?: number, end?: number): void;
 
 	/**
-	 * Moves the specified items to the start of the sequence.
+	 * Moves the specified item to the start of the list.
+	 * @param sourceIndex - The index of the item to move.
+	 * @throws Throws if `sourceIndex` is not in the range [0, `list.length`).
+	 */
+	moveToStart(sourceIndex: number): void;
+
+	/**
+	 * Moves the specified item to the start of the list.
+	 * @param sourceIndex - The index of the item to move.
+	 * @param source - The source list to move the item out of.
+	 * @throws Throws if `sourceIndex` is not in the range [0, `list.length`).
+	 */
+	moveToStart(sourceIndex: number, source: SharedTreeList<AllowedTypes>): void;
+
+	/**
+	 * Moves the specified item to the end of the list.
+	 * @param sourceIndex - The index of the item to move.
+	 * @throws Throws if `sourceIndex` is not in the range [0, `list.length`).
+	 */
+	moveToEnd(sourceIndex: number): void;
+
+	/**
+	 * Moves the specified item to the end of the list.
+	 * @param sourceIndex - The index of the item to move.
+	 * @param source - The source list to move the item out of.
+	 * @throws Throws if `sourceIndex` is not in the range [0, `list.length`).
+	 */
+	moveToEnd(sourceIndex: number, source: SharedTreeList<AllowedTypes>): void;
+
+	/**
+	 * Moves the specified item to the desired location in the list.
+	 * @param index - The index to move the item to.
+	 * This is relative to the list before moving the source item.
```

Comment:
This "relative" phrasing bothers me a little bit. When we say an index is relative to something we generally mean that it is an offset.
How about:
```suggestion
	 * This is based on the state of the list before moving the source item.
```

Reply:
Much better, thanks.